src/tasks/move_directory.rs

```diff
@@ -0,0 +1,45 @@
+use std::{fs, path};
+use tracing_batteries::prelude::tracing;
+use crate::core::{Core, Target};
+use crate::{errors};
+use crate::tasks::{Task};
+
+pub struct MoveDirectory {
+    pub new_path: path::PathBuf,
+}
+
+#[async_trait::async_trait]
+impl Task for MoveDirectory {
+    #[tracing::instrument(name = "task:git_clone(repo)", err, skip(self, _core))]
+    async fn apply_repo(&self, _core: &Core, repo: &crate::core::Repo) -> Result<(), crate::core::Error> {
+        if !repo.exists() {
+            return Err(errors::user(
+                "Repository does not exist.",
+                "Provide the name of the repository you wish to rename.",
+            ))
+        }
+
+        fs::rename(repo.path.clone(), self.new_path.clone()).map_err(|err| {
+            errors::user_with_internal(
+                &format!(
+                    "Could not rename the repository directory '{}' to '{}' due to an OS-level error.",
+                    repo.path.display(),
+                    self.new_path.display()
+                ),
+                "Check that Git-Tool has permission to create this directory and any missing parent directories.",
+                err,
+            )
+        })?;
+
+        Ok(())
+    }
+
+    #[tracing::instrument(name = "task:git_rename(scratchpad)", err, skip(self, _core))]
+    async fn apply_scratchpad(
+        &self,
+        _core: &Core,
+        _scratch: &crate::core::Scratchpad,
+    ) -> Result<(), crate::core::Error> {
+        Ok(())
```

Comment:
It would be cool to support this for scratchpads as well - being able to change the name of a scratchpad seems like a neat ability (although I'm happy to add this in a follow-up PR).

Reply:
Yes, indeed. Wanted to get the git repo part done first, but I have firm plans adding this to scratch pads as well